core/src/main/java/com/backblaze/b2/client/B2LargeFileStorer.java

```diff
@@ -77,6 +86,28 @@
         return partStorers;
     }
 
+    private static List<Long> computeStartingBytePositions(List<B2PartStorer> partStorers) {
+        final List<Long> startingPositions = new ArrayList<>();
+
+        long cursor = 0;
+        try {
+            for (final B2PartStorer partStorer : partStorers) {
+                startingPositions.add(cursor);
+                cursor += partStorer.getPartSizeOrThrow();
+            }
+        } catch (B2CannotComputeException e) {
+            while (startingPositions.size() < partStorers.size()) {
+                startingPositions.add(B2UploadProgress.UNKNOWN_PART_START_BYTE);
+            }
+        }
+
+        return startingPositions;
+    }
+
+    long getStartByte(int partNumber) {
```

Comment:
"...OrUnknown"? or maybe just a comment to point out that it might return UNKNOWN_PART_START_BYTE?

Reply:
Done.